From 9522f31c41422621e0dd8ef56ba9dba24ec5f9b2 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Wed, 19 Nov 2025 18:45:14 +0800
Subject: [PATCH 045/268] r8169: Remove the check of a register to enable ASPM
 on Radxa Dragon Q6A

This will reduce the amount of PCIe AERs generated after system resume.
Exact reasons of the AER still need to be investigated.

Signed-off-by: Xilin Wu <sophon@radxa.com>
---
 drivers/net/ethernet/realtek/r8169_main.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/net/ethernet/realtek/r8169_main.c
+++ b/drivers/net/ethernet/realtek/r8169_main.c
@@ -6221,8 +6221,7 @@ done:
 /* register is set if system vendor successfully tested ASPM 1.2 */
 static bool rtl_aspm_is_safe(struct rtl8169_private *tp)
 {
-	if (tp->mac_version >= RTL_GIGA_MAC_VER_46 &&
-	    r8168_mac_ocp_read(tp, 0xc0b2) & 0xf)
+	if (tp->mac_version >= RTL_GIGA_MAC_VER_46)
 		return true;
 
 	return false;
